bson/binary.py

```diff
@@ -513,6 +522,12 @@ def as_vector(self) -> BinaryVector:
             dtype_format = "B"
             format_string = f"<{n_values}{dtype_format}"
             unpacked_uint8s = list(struct.unpack_from(format_string, self, position))
+            if padding and not n_values:
+                raise ValueError("Corrupt data. Vector has a padding P, but no data.")
+            if padding and n_values and not (unpacked_uint8s[-1] & ((1 << padding) - 1)) == 0:
+                raise ValueError(
```

Comment:
Are these breaking changes for existing data?

Reply:
They are, yeah. But only for a case that vector search doesn't use. I feel  confident that it won't affect anyone now.

Reply:
Please add a changelog entry to explain this because it's a breaking change.